### DISCUSSION TITLE
Multiply Instead of Repeats - Option to Train on a Portion of Images Per Epoch

### DISCUSSION THREAD

**Discussion by a-l-e-x-d-s-9** [bmaltais/kohya_ss#1514] [Ideas]
I'm familiar with the [EveryDread2](https://github.com/victorchall/EveryDream2trainer) (ED2) trainer, which provides an option to specify any portion of images from a folder. Instead of simply defining the number of repeats for a folder, you can set a 0.5 multiplier for a folder, and it will train on half of the images, selected randomly each epoch. You can also use a 1.6 multiplier or any other positive number. Negative number will discard the folder, which also important.
This approach not only includes the concept of repeats but also offers greater flexibility. It's a more effective way to balance between subjects without the need to discard any images.
I've utilized this feature in every training session with ED2, and I'm convinced that it's a crucial feature for the Kohya trainer as well.
If anyone is familiar with the Kohya codebase, could you please assess whether it's feasible to implement this feature in a reasonable manner? Your feedback would be appreciated.

**Comment by morozig** [comment]
Guys, any news on that? That's a super useful feature.
